client/core/core.go

```diff
@@ -2478,6 +2814,57 @@ func (c *Core) Login(pw []byte) (*LoginResult, error) {
 	return result, nil
 }
 
+// upgradeCoreToV1 performs an authenticated upgrade.
+func (c *Core) upgradeCoreToV1(pw []byte) error {
+	const newVersion = 1
+
+	c.log.Infof("Upgrading to core version 1")
+
+	oldKeyParams, err := c.db.LegacyKeyParams()
+	if err != nil {
+		return err
+	}
+	oldCrypter, err := c.reCrypter(pw, oldKeyParams)
+	if err != nil {
+		return fmt.Errorf("legacy encryption key deserialization error: %w", err)
+	}
+
+	newCrypter, creds, err := c.generateCredentials(pw, nil)
+	if err != nil {
+		return err
+	}
```

Comment:
Maybe it's just me, but I'd prefer my app to print the seed to stdout right at this point.  Users should absolutely grab the seed and back it up, and making them call an RPC or go to the /settings page, or (for Go consumers) call the `ExportSeed` method seems like enough of a hurdle that upgrading users will be unlikely to backup their new seed.

Reply:
I was trying to avoid dumping the seed in the terminal history, especially since users are likely not using a CLI and wouldn't be considering the risks involved with that.

Reply:
Then since the user is getting an automatic upgrade when they login, I think there needs to be a notification with warning level severity when the upgrade is complete that they should go to /settings to export/backup their seed.